src/libs/pluginprocess/pluginprocess.c

```diff
@@ -396,26 +470,71 @@ int elektraPluginClose (Plugin * handle, Key * errorKey)
  * @retval 0 if the data structure is still used
  * @ingroup processplugin
  **/
-int elektraPluginProcessClose (ElektraPluginProcess * pp, Key * errorKey)
+ElektraPluginProcessCloseResult elektraPluginProcessClose (ElektraPluginProcess * pp, Key * errorKey)
 {
 	pp->counter = pp->counter - 1;
+	int result = elektraPluginProcessSend (pp, ELEKTRA_PLUGINPROCESS_CLOSE, NULL, errorKey);
 	int done = pp->counter <= 0;
 	if (done) cleanupPluginData (pp, errorKey);
-	return done;
+	ElektraPluginProcessCloseResult closeResult = { result, done };
+	return closeResult;
 }
 
 /** Store a pointer to any plugin related data that is being executed inside an own process.
+ *
+ * This is required in case additional arbitrary plugin data should be stored. Pluginprocess
+ * has to be stored using elektraPluginSetData. Plugin data for the child process
+ * has to be stored using this function like
+ * @code
+int elektraPluginOpen (Plugin * handle, Key * errorKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (pp == NULL)
+	{
+		if ((pp = elektraPluginProcessInit (errorKey)) == NULL) return ELEKTRA_PLUGIN_STATUS_ERROR;
+		ArbitraryPluginData * data = // initialize your plugin data
+		elektraPluginProcessSetData (handle, data);
```

Comment:
Ahh, thank you. That makes it clear how to use it.

Reply:
yes that was the main mistake i've seen in your python plugin, i am currently trying to fix it